src/modules/accommodations/components/reservation-card.tsx

```diff
@@ -1,172 +1,198 @@
 
-interface ReservationCardProps {
-  pricePerNight: number;
-  cleaningFee: number;
-  serviceFee: number;
-  maxGuests: number | null;
-  availableFrom: string | null;
-  availableTo: string | null;
-}
+dayjs.extend(isBetween);
 
 const styles = {
-  cardContainer: (theme: any) => ({
-    border: '1px solid',
-    borderColor: theme.palette.primary.main,
+  container: {
+    border: '1px solid #ddd',
     borderRadius: '8px',
     padding: '16px',
     maxWidth: '400px',
-    boxShadow: '0px 2px 4px rgba(0, 0, 0, 0.1)',
-  }),
-  button: (theme: any, isEnabled: boolean) => ({
-    backgroundColor: isEnabled ? theme.palette.secondary.main : theme.palette.action.disabled,
-    color: 'white',
-    fontWeight: 'bold',
-    marginBottom: '16px',
-    height: '56px',
-    '&:hover': {
-      backgroundColor: isEnabled ? theme.palette.secondary.dark : theme.palette.action.disabled,
-    },
-  }),
-  textField: {
-    flex: 1,
-  },
-  section: {
-    display: 'flex',
-    justifyContent: 'space-between',
-    marginBottom: '8px',
-  },
-  headingPrice: {
-    fontWeight: 'bold',
-    marginBottom: '16px',
+    boxShadow: '0 2px 8px rgba(0, 0, 0, 0.1)',
   },
-  flexBox: {
+  datePickerContainer: {
     display: 'flex',
     justifyContent: 'space-between',
     alignItems: 'center',
     marginBottom: '16px',
   },
-  guestsDropdown: {
-    marginBottom: '16px',
-  },
-  notChargedText: (theme: any) => ({
-    textAlign: 'center',
-    color: theme.palette.grey[600],
+  button: (isButtonEnabled: boolean) => ({
+    backgroundColor: isButtonEnabled ? '#FF385C' : '#ddd',
+    color: isButtonEnabled ? '#fff' : '#aaa',
+    fontWeight: 'bold',
+    padding: '12px 0',
+    textTransform: 'none',
+    fontSize: '16px',
     marginBottom: '16px',
   }),
-  divider: {
-    marginBottom: '16px',
-  },
-  totalText: {
-    fontWeight: 'bold',
-  },
 };
 
-export const ReservationCard: React.FC<ReservationCardProps> = ({
+interface ReservationCardProps {
+  pricePerNight: number;
+  cleaningFee: number;
+  serviceFee: number;
+  maxGuests: number | null;
+  availableFrom: string | null;
+  availableTo: string | null;
+  accommodationId: string;
+  bookings: { startDate: string; endDate: string }[];
+}
+
+const ReservationCard: React.FC<ReservationCardProps> = ({
   pricePerNight,
   cleaningFee,
   serviceFee,
   maxGuests,
-  availableFrom,
   availableTo,
+  accommodationId,
+  bookings,
 }) => {
-  const theme = useTheme();
-
-  const [checkIn, setCheckIn] = useState<string>('');
-  const [checkOut, setCheckOut] = useState<string>('');
+  const [checkIn, setCheckIn] = useState<Dayjs | null>(null);
+  const [checkOut, setCheckOut] = useState<Dayjs | null>(null);
   const [guests, setGuests] = useState<number>(1);
+  const [total, setTotal] = useState<number>(0);
   const [isButtonEnabled, setIsButtonEnabled] = useState<boolean>(false);
 
+  const todayDate = dayjs();
+
+  const isDateUnavailable = (date: Dayjs): boolean => {
```

Comment:
Please move this to reservation card util. Also, please move ```shared/utils/reservatoin-card.util.ts``` to ```modules/accommodations/reservation-card.util.ts``` (and please correct the typo)